### PR TITLE
Refactor postcss-merge-longhand

### PR DESCRIPTION
**Pull request by andyjansson**
Added more precise parsing of `width` `style` `color`, which in turn has opened the door to parse borders with custom properties. Also added an additional border optimisation, increasing compression.

### CODE REVIEW

File: packages/postcss-merge-longhand/src/lib/validateWsc.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    if (/#([0-9a-zA-Z]{6}|[0-9a-zA-Z]{3})/.test(value)) {
+        return true;
+    }
+    if (value === 'transparent') {
+        return true;
+    }
+
+    if (value === 'currentColor') {
```
Comment (evilebottnawi) [inline packages/postcss-merge-longhand/src/lib/validateWsc.js]: All values from list can be in UPPERCASE format, css is case insensitivity (in many cases), don't forget this

File: packages/postcss-merge-longhand/src/lib/parseWsc.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import {list} from 'postcss';
+import {isWidth, isStyle, isColor} from './validateWsc';
+
+export default function parseWsc (value) {
+    if (value === 'none' || value === 'none none' || value === 'none none currentColor') {
```
Comment (evilebottnawi) [inline packages/postcss-merge-longhand/src/lib/parseWsc.js]: Also can be case insensitivity

File: packages/postcss-merge-longhand/src/lib/decl/borders.js [hunk 3]
```diff
@@ -47,49 +50,18 @@ function getLevel (prop) {
     }
 }
 
-const widths = ['thin', 'medium', 'thick'];
-const styles = ['none', 'hidden', 'dotted', 'dashed', 'solid', 'double', 'groove', 'ridge', 'inset', 'outset'];
+const isValueCustomProp = value => value.search(/var\s*\(\s*--/i);
```
Comment (evilebottnawi) [inline packages/postcss-merge-longhand/src/lib/decl/borders.js]: var can be case insensitivity, also i think we can avoid check on `--` in regex

**Comment by evilebottnawi** [comment]
@andyjansson Good work, some notes above

**Comment by andyjansson** [comment]
I guess.

**Pull request merged**